services/polly/src/main/java/software/amazon/awssdk/services/polly/internal/presigner/DefaultPollyPresigner.java

```diff
@@ -124,22 +136,28 @@ public void close() {
         IoUtils.closeIfCloseable(credentialsProvider, null);
     }
 
+    // For testing
+    static PollyPresigner.Builder builder(Clock signingClock) {
```

Comment:
> Minor: instead of exposing setter method for testing, can we use dependency injection for testing, something like below

Sure, I can change it, but just to clarify, the setter is not ever exposed as is not part of the type interface returned by this method.